src/python/WMCore/WMSpec/Steps/Executors/DQMUpload.py

```diff
@@ -293,3 +309,229 @@ def upload(self, url, args, filename):
+
+        ident = "WMAgent python/%d.%d.%d" % sys.version_info[:3]
+        uploadProxy = self.step.upload.proxy or os.environ.get('X509_USER_PROXY', None)
+        logging.info("Using proxy file: %s", uploadProxy)
+        logging.info("Using CA certificate path: %s", os.environ.get('X509_CERT_DIR'))
+
+        msg = "HTTP Register POST arguments: %s\n" % args
+        logging.info(msg)
+
+        handler = HTTPSAuthHandler(key=uploadProxy, cert=uploadProxy)
+        opener = OpenerDirector()
+        opener.add_handler(handler)
+
+        # setup the request object
+        url = decodeBytesToUnicode(registerURL)
+        datareq = Request(url)
+        datareq.add_header('Accept-encoding', 'gzip')
+        datareq.add_header('User-agent', ident)
+        datareq.data = encodeUnicodeToBytes(json.dumps(args))
+
+        if 'https://' in url:
+            result = opener.open(datareq)
+        else:
+            opener.add_handler(ProxyHandler({}))
+            result = opener.open(datareq)
+
+        return result
+    
+    def _registerAnalysisFile(self, stepName, analysisFile):
+        """
+        Register a file to new DQM GUi
+        - https://github.com/cms-DQM/dqmgui
+        HTTP request body:
+            [{"dataset": "/a/b/c", "run": "123456", "lumi": "0", "file": "/eos/cms/store/group/comm_dqm/DQMGUI_data/location/file.root", "fileformat": 1}]
+        Set lumis to 0, as this is reproducing current per Run based root files
+        - https://github.com/dmwm/WMCore/issues/10287#issuecomment-1052558061
+        DQM root files produced by Harvesting prosessing are type 1 (plain ROOT)
+        - https://github.com/cms-DQM/dqmgui#file-formats
+        """
+        # Get task description
+        job = Bootstrap.loadJobDefinition()
+        jobBag = job.getBaggage()
+        datasetName = job.get('inputDataset', None)
+
+        # Get runNumber, depending on the harvesting job mode (byRun vs multiRun)
+        multiRun = getattr(jobBag, "multiRun", False)
+        forceRunNumber = getattr(jobBag, "forceRunNumber", 999999)
+        if multiRun:
+            runNumber = forceRunNumber
+        else:
+            try:
+                runNumber = list(job['mask']['runAndLumis'].keys())[0]
+            except Exception as ex:
+                msg = "Error while retrieving run number from job definition:\n %s" % str(ex)
+                raise WMExecutionFailure(70320, "DQMUploadFailure", msg)    
+
+        args = {}
+        args['file'] = self.registerEOSPrefix + \
+                self.getAnalysisFileLFN(analysisFile)
+        args['dataset'] = datasetName
+        args['run'] = runNumber
+        args['lumi'] = 0
+        args['fileformat'] = 1
+
+        msg = "HTTP Upload is about to start:\n"
+        msg += " => URL: %s\n" % self.registerURL
+        msg += " => Filename: %s\n" % args['file']
+        logging.info(msg)
+
+        try:
+            logException = True
+            for registerURL in self.registerURL.split(';'):
```

Comment:
We should contact the DQM Gui and see whether the file registration should be done against a single server (for each workflow). If so, I'm very much in favor of removing this. Actually, it was added to the DQMUpload step many many years ago, I am not even sure whether it's still required and useful to anyone.

If we can remove the multi url manipulation, then we can likely try to remove this `logException` flag as well.

Reply:
Good point! I just left a comment in the issue asking the DQM folks.